### PR TITLE
152 backed api endpoints for workout collection and bugs

### LINKED ISSUE #152: API endpoints

### PR DESCRIPTION
**Pull request by bigboidanwithacan**
# What type of PR is this? (Check all that apply)

- [ ] ✨**feature**: Introduces completely new code or new features.
- [x] 🐛**fix**: Implements changes that fix a bug. Ideally, reference an issue if present.
- [ ] ♻️**refactor**: Includes any code-related change that is neither a fix nor a feature.
- [ ] ✅**build**: Encompasses all changes related to the build of the software, including changes to dependencies or the addition of new ones.
- [ ] ⚡️**test**: Pertains to all changes regarding tests, whether adding new tests or modifying existing ones.
- [ ] 🚰**ci**: Involves all changes related to the configuration of continuous integration, such as GitHub Actions or other CI systems.
- [ ] 📚**docs**: Includes all changes to documentation, such as README files, or any other documentation present in the repository.
- [ ] 🗑️**chore**: Captures all changes to the repository that do not fit into the above categories.

# Description

What did you change? How did you change it?

Fix for two functions inside the API endpoint for a workout.
When a new workout is created, instead of creating new exercises also, it now links exercises in the database to the new workout.
deleteExerciseFromWorkout also successfully disconnects an exercise from a workout.

Made changes to the schema also
changed log name to WorkoutLog
added a customExercies field to the User model
changed the relation workout and exercise had to each other to let a single exercise connect to multiple workouts
 
close #152 

# Tests

How was this tested?

- [ ] Unit tests
- [ ] Integration tests
- [ ] E2E tests
- [x] Manual tests
- [ ] Tests were NOT needed
- [ ] Other (explain below)

# [Optional] Screenshots
image
image

# Documentation

- [ ] Added to README.me
- [ ] Seperate document
- [x] NO documentation needed

Link to external documentation:

# [Optional] Are there any post-deployment tasks we need to perform?

### CODE REVIEW

File: packages/db/prisma/schema.prisma [hunk 1]
```diff
@@ -91,10 +91,10 @@ model User {
   username             String         @unique @db.Char(20)
   name                 String?
   status               UserStatus?
-  streak               Int            @default(0)
```
Comment (mva919) [inline packages/db/prisma/schema.prisma:94]: why was this deleted?

File: packages/db/src/scripts/seed.ts [hunk 2]
```diff
@@ -58,14 +58,14 @@ const loaddb = async () => {
...
-    logger('delete', 'log')
+    await prisma.workoutLog.deleteMany()
+    logger('delete', 'workoutLog')
 
     await prisma.workout.deleteMany()
     logger('delete', 'workout')
 
-    await prisma.log.deleteMany()
-    logger('delete', 'log')
+    // await prisma.log.deleteMany()
```
Comment (taylorfinelli) [inline packages/db/src/scripts/seed.ts:67]: Why was this commented out?

Comment (taylorfinelli) [reply packages/db/src/scripts/seed.ts:67]: If it is because log changed to workoutLog then can we just rename it?

Comment (taylorfinelli) [reply packages/db/src/scripts/seed.ts:67]: If it is no longer needed can we just delete it?

File: packages/db/prisma/schema.prisma (hunk 1 above)
Comment (mva919) [reply packages/db/prisma/schema.prisma:94]: WHY!!!???

Comment (taylorfinelli) [reply packages/db/prisma/schema.prisma:94]: YOU HEARD THE MAN

**Comment by thienguen** [comment]
not fall over cycle 9. Implementing fix ...

File: packages/db/prisma/schema.prisma (hunk 1 above)
Comment (bigboidanwithacan) [reply packages/db/prisma/schema.prisma:94]: My fault OG

File: packages/db/src/scripts/seed.ts (hunk 2 above)
Comment (bigboidanwithacan) [reply packages/db/src/scripts/seed.ts:67]: Hey, sorry about that; there were 2 delete statements for log(workoutLog now) before, and I commented one out to see if there was a need for it when I ran it. It has been removed now.

File: packages/db/prisma/schema.prisma [hunk 3]
```diff
@@ -175,14 +176,15 @@ model Workout {
 // * GROUP: Group message
 // ******************************************************************************************
 model Exercise {
-  id        Int      @id @unique @default(autoincrement())
-  name      String   @db.Char(100)
+  id        Int       @id @unique @default(autoincrement())
+  name      String    @db.Char(100)
   note      String?
   sets      Set[]
```
Comment (thienguen) [inline packages/db/prisma/schema.prisma:182]: blame @bigboidanwithacan
